tests/pyreverse/test_diadefs.py

```diff
@@ -22,25 +22,30 @@
 # pylint: disable=redefined-outer-name
 import sys
 from pathlib import Path
+from typing import Callable, Dict, List, Tuple
 
 import pytest
 from astroid import nodes
+from conftest import PyreverseConfig  # type: ignore #pylint: disable=no-name-in-module
```

Comment:
I couldn't get this import to pass for `mypy`.

As this is a test the import from `conftest` works but pylint throws an error.

Reply:
``mypy`` cannot deal with modules that have the same name and are inside a "non-package" directory tree (i.e. no ``__init__.py``). I already had to exclude every ``conftest.py`` except the top-level one in the pre-commit configuration for ``mypy`` because of this.
A solution to this would be to add ``__init__.py`` files to the tests folders and turn them into packages.
But this is [warned against in the pytest docs](https://docs.pytest.org/en/6.2.x/goodpractices.html#tests-outside-application-code) for repositories that use a non-src layout in combination with ``tox``, which is exactly the case for pylint.

Reply:
So this is the best way to do this? Or would you suggest another solution?

Reply:
The only workaround I could think of is extracting the ``PyreverseConfig`` class in a separate module, from which both the ``conftest.py`` and ``test_diadefs.py`` import it. But as this is "just" test code and not production code, I don't think it is worth it. 
So I would say leave it as it is. 
(Funnily enough, the import is only necessary because ``PyreverseConfig`` is itself used as a type hint 😁)

Reply:
> (Funnily enough, the import is only necessary because PyreverseConfig is itself used as a type hint 😁)

We could use ``if typing.TYPE_CHECKING`` for that. But the issue with conftest.py and mypy is between pytest design and mypy's design and we can't help it. I agree with @DudeNr33 it's not worth a refactor.